pipelinewise/fastsync/commons/tap_postgres.py

```diff
@@ -287,6 +288,33 @@ def get_current_lsn(self) -> int:
                 )
         return parse_lsn(result[0]['current_lsn'])
 
+    def get_confirmed_flush_lsn(self) -> int:
+        """
+        Get the last flushed LSN for the replication slot.
+
+        For Postgres <9.6 this defaults to the restart_lsn as confirmed_flush_lsn
+        is not availiable.
+        """
+        slot_name = self.__get_slot_name(
+            self.primary_host_conn,
+            self.connection_config['dbname'],
+            self.connection_config['tap_id'],
+        )
+
+        res = self.primary_host_query(
+            textwrap.dedent(
+                f"""SELECT *
+                    FROM pg_replication_slots
+                    WHERE slot_name = '{slot_name}'
+                    AND plugin = 'wal2json'
+                    AND slot_type = 'logical'"""
+            )
```

Comment:
I don't think dedent is going to dedent this:
```python
In [1]: import textwrap

In [2]: textwrap.dedent("Foo\n    Bar")
Out[2]: 'Foo\n    Bar'

In [3]: textwrap.dedent("    Foo\n    Bar")
Out[3]: 'Foo\nBar'
```

Reply:
https://github.com/thread/pipelinewise/pull/25/commits/783e28ee9068ea6516f65a1895dc2d22e3830a29 appears to address this.

Note however that `textwrap.dedent` _can_ handle leading non-indented newlines:
``` python
In [6]: textwrap.dedent("\n    Foo\n    Bar")
Out[6]: '\nFoo\nBar'
```
So the leading `\` in the current version of this is not needed.

Reply:
You end up with an extra `\n` at the start then though.